// express-vue-renderer/lib/models/defaults.js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var LRU = require('lru-cache');
var path = require('path');
var deepmerge = require('deepmerge');
var Layout = require('./layout');
var options = {
    max: 500,
    maxAge: 1000 * 60 * 60
};
var lruCache = LRU(options);

var Defaults = function () {
    function Defaults() {
        var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

        _classCallCheck(this, Defaults);

        this.cache = lruCache;
        this.options = options;
        this.layout = new Layout.Layout(options.layout);

        if (options.rootPath) {
            this.rootPath = path.resolve(options.rootPath);
        }
        if (options.component) {
            this.component = options.component;
        }

        if (options.vue) {
            this.vue = options.vue;
        } else {
            this.vue = {};
        }
        if (options.data) {
            this.data = options.data;
        } else {
            this.data = {};
        }
    }

    _createClass(Defaults, [{
        key: 'mergeVueObject',
        value: function mergeVueObject(newVueObject) {
            this.vue = deepmerge(this.vue, newVueObject);
        }
    }, {
        key: 'mergeDataObject',
        value: function mergeDataObject(newDataObject) {
            this.data = deepmerge(this.data, newDataObject);
        }
    }]);

    return Defaults;
}();

module.exports = Defaults;
