// express-vue-renderer/lib/models/layout.js
'use strict';

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var HTML = function HTML() {
    var html = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    _classCallCheck(this, HTML);

    this.start = html.start ? html.start : '<!DOCTYPE html><html>';
    this.end = html.end ? html.end : '</html>';
};

var Body = function Body() {
    var body = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    _classCallCheck(this, Body);

    this.start = body.start ? body.start : '<body>';
    this.end = body.end ? body.end : '</body>';
};

var Template = function Template() {
    var template = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    _classCallCheck(this, Template);

    this.start = template.start ? template.start : '<div id="app">';
    this.end = template.end ? template.end : '</div>';
};

var Layout = function Layout() {
    var obj = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    _classCallCheck(this, Layout);

    this.html = new HTML(obj.html);
    this.body = new Body(obj.body);
    this.template = new Template(obj.template);
};

module.exports.Layout = Layout;
module.exports.HTML = HTML;
module.exports.Body = Body;
module.exports.Template = Template;
